lib/mnemosyne/probes/redis/command.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Mnemosyne
+  module Probes
+    module Redis
+      module Command
+        class Probe < ::Mnemosyne::Probe
+          def setup
+            ::Redis::Client.prepend ClientPatch
+          end
+
+          module ClientPatch
+            def process(commands)
+              trace = ::Mnemosyne::Instrumenter.current_trace
+              return super unless trace
+
+              span = ::Mnemosyne::Span.new 'db.command.redis',
+                meta: extract_span_meta(commands)
+
+              span.start!
+
+              super.tap do |retval|
+                span.meta[:error] = retval.message if retval.is_a?(::Redis::CommandError)
+
+                trace << span.finish!
+              end
+            end
+
+            private
+
+            def extract_span_meta(commands)
+              {
+                server: id,
+
+                # Each command is an array, consisting of the command name and any
+                # arguments. We are only interested in the command name.
+                commands: extract_command_names(commands),
+
+                # If there are multiple commands, that must mean they were pipelined
+                # (i.e. run in parallel).
+                pipelined: commands.length > 1
+              }
```

Comment:
Do we want to collect Redis keys too?

Reply:
Okay, I just whitelisted the name argument for `SET` and `GET`. For simple commands, this is fairly extensible, so it should be easy to add more later.

Does this (i.e. the last commit) look good?

Reply:
It does. Thanks!